x-pack/plugins/security_solution/public/actions/utils.ts

```diff
@@ -35,3 +36,13 @@ export const isCountField = (
 ) => {
   return fieldType === 'number' && sourceParamType === 'value_count';
 };
+
+const SUPPORTED_LENS_TYPES = new Set([
+  KBN_FIELD_TYPES.STRING,
+  KBN_FIELD_TYPES.BOOLEAN,
+  KBN_FIELD_TYPES.NUMBER,
+  KBN_FIELD_TYPES.IP,
+]);
+
+export const isLensSupportedType = (fieldType: string | undefined) =>
+  fieldType ? SUPPORTED_LENS_TYPES.has(fieldType as KBN_FIELD_TYPES) : false;
```

Comment:
The `isLensSupportedType` is only used by showTopN _cellAction_, it should not be defined as a common util function. Can we move it to the showTopN action file that uses it?

Also, it is a bit confusing having `isLensEmbeddable` util function used by all _lensActions_, and the `isLensSupportedType` which is not used by _lensActions_, in the same place.

Reply:
Good catch! I moved it from `public/actions/utils` to `public/common/utils` because I found out that detection had implemented the same function.